lib/toggle_switch.dart

```diff
@@ -427,6 +431,10 @@ class _ToggleSwitchState extends State<ToggleSwitch>
   void _handleOnTap(int index) async {
     bool notifyNull = false;
     if (widget.changeOnTap) {
+      if (widget.disableSameStatusToggle == true &&
+          widget.initialLabelIndex == index) {
+        return;
+      }
```

Comment:
I'm no maintainer, but why not just put these in the original if?
```suggestion
    if (widget.changeOnTap && (!widget.disableSameStatusToggle && widget.initialLabelIndex != index)) {
```

Reply:
I have committed to the changes you suggested. For boolean == true the disableSameStatusToggle is a nullable boolean so for null safety, I am doing this.